quil-rs/src/expression/simplification/by_hand.rs

```diff
@@ -1,13 +1,20 @@
-/// Complex machinery for simplifying [`Expression`]s.
+//! Complex machinery for simplifying [`Expression`]s.
+
+use std::{cmp::min_by_key, collections::HashMap};
+
+use internment::ArcIntern;
+
 use crate::expression::{
-    imag, real, Expression, ExpressionFunction, FunctionCallExpression, InfixExpression,
-    InfixOperator, PrefixExpression, PrefixOperator,
+    calculate_function, calculate_infix, interned, real, Expression, ExpressionFunction,
+    FunctionCallExpression, InfixExpression, InfixOperator, PrefixExpression, PrefixOperator,
 };
-use std::cmp::min_by_key;
 
 /// Simplify an [`Expression`].
-pub(super) fn run(expression: &Expression) -> Expression {
-    simplify(expression, LIMIT)
+pub fn run(expression: &Expression) -> Expression {
```

Comment:
What's the motivation for changing this to fully `pub`?

Reply:
If (!) I understood correctly, this is `pub` _relative to the containing module_, which is not `pub`, so it's just about letting us `pub(super) use` it in `simplification/mod.rs`